datafusion/core/src/execution/context.rs

```diff
@@ -816,7 +809,7 @@ impl SessionContext {
             name,
             table_path,
             listing_options,
-            options.schema,
+            options.schema.map(|s| Arc::new(s.to_owned())),
```

Comment:
I don't think it really matters but you can probably doe

Reply:
Related to an API change (CSV and JSON - AVRO unification issue).

Reply:
`register_avro` and `register_json` methods take `AvroReadOptions` and `NdJsonReadOptions` as arguments respectively. When we try to unify them with `CsvReadOptions`, this change is obliged since 

```
pub struct CsvReadOptions<'a> {
  /// An optional schema representing the CSV files. If None, CSV reader will try to infer it
  /// based on data in file.
  pub schema: Option<&'a Schema>,
}
```
and we do not want to break the CSV API. If it is OK to break CSV API, your suggestions are incredibly logical.